### ISSUE TITLE
Add a proper implementation for

### ISSUE THREAD

**Issue by antgamdia** [kubeapps/kubeapps#3854]

**Comment by antgamdia** [comment]
Quick update:

using the `GroupVersionKind` to `GroupVersionResource` approach that Michael came up with, plus the `ServerPreferredResources` from k8s, it is possible to fetch every possible resource within a given kubernetes cluster installation.

Tests are still pending, tomorrow I'll send a PR, but more or less this is the output:

![image](https://user-images.githubusercontent.com/11535726/148853362-b57c8746-6440-4ae3-9e18-9772f324f5ec.png)


Question for @absoludity: some carvel packages (ie, harbor or cert-manager) install their generated resources in a different namespace from the one in which the `PackageInstall`. As a result, the "resources" plugin is not returning those resources that are in other namespaces. I guess something is wrong here, no? I'll file an issue tomorrow.

Extra: I've also been working on removing the ApplicationStatusContainer (ie, just using hooks), but since I wasn't able to get it to work with the carvel packages I had installed so far, it's still wip.

**Comment by absoludity** [comment]
> Question for @absoludity: some carvel packages (ie, harbor or cert-manager) install their generated resources in a different namespace from the one in which the `PackageInstall`. As a result, the "resources" plugin is not returning those resources that are in other namespaces. I guess something is wrong here, no? I'll file an issue tomorrow.

Yeah, this is something we've not had to deal with for Helm, but carvel and flux (from memory) will enable. As long as we know from the `PackageInstall` where the resources should be (as we can trust the package install), we can update to return those, I think.

**Comment by absoludity** [comment]
> > Question for @absoludity: some carvel packages (ie, harbor or cert-manager) install their generated resources in a different namespace from the one in which the `PackageInstall`. As a result, the "resources" plugin is not returning those resources that are in other namespaces. I guess something is wrong here, no? I'll file an issue tomorrow.
> 
> Yeah, this is something we've not had to deal with for Helm, but carvel and flux (from memory) will enable. As long as we know from the `PackageInstall` where the resources should be (as we can trust the package install), we can update to return those, I think.

I just took a brief look at this issue after you mentioned @antgamdia it in the standup:

- The [ResourceRef message](https://github.com/kubeapps/kubeapps/blob/master/cmd/kubeapps-apis/proto/kubeappsapis/core/packages/v1alpha1/packages.proto#L924-L928) already explicitly supports references for resources that are in other namespaces than the installed package context.
- The tests for the [resources plugin already have a case for this](https://github.com/kubeapps/kubeapps/blob/master/cmd/kubeapps-apis/plugins/resources/v1alpha1/server_test.go#L318).

but, that said, I wasn't able (yet) to test the streaming requests and as it turns out, it was using the installed package context namespace, rather than that of the resource.

This one-liner (#4057) should fix it, though I'm keen to get the tests working for streaming requests to the fake k8s client too (need to add a watch reactor or something, which I think Greg may have done in some of his code already).
